ip/flac.c

```diff
@@ -427,11 +415,14 @@ static int flac_seek(struct input_plugin_data *ip_data, double offset)
 
 	sample = (uint64_t)(offset * (double)sf_get_rate(ip_data->sf) + 0.5);
 	if (!F(seek_absolute)(priv->dec, sample)) {
+		if (F(get_state(priv->dec)) == FLAC__STREAM_DECODER_SEEK_ERROR) {
+			if (!F(flush)(priv->dec))
+				d_print("failed to flush\n");
+			priv->buf_rpos = 0;
+			priv->buf_wpos = 0;
+		}
 		return -IP_ERROR_ERRNO;
 	}
-	priv->ignore_next_write = 1;
-	priv->buf_rpos = 0;
-	priv->buf_wpos = 0;
```

Comment:
I think the original author is mislead by flac doc, which mentioned the stream may contain a partial block when seeking. So he tried to reset that block. But what flac doc really means is you won't get the entire block as you normally do when using process_single, becauce the sample may live in the middle of a flac block (which contains various samples). But flac ensures you that you won't get partial samples. You'll always get right data stream when doing seek_absolute. There's no need to reset at all.